pkg/provider/azure_vmssflex_cache.go

```diff
@@ -106,8 +107,8 @@ func (fs *FlexScaleSet) newVmssFlexVMCache() (*azcache.TimedCache, error) {
 
 		for i := range vms {
 			vm := vms[i]
-			if vm.Name != nil {
-				cached, ok := localCache.Load(*vm.Name)
+			if vm.OsProfile != nil && vm.OsProfile.ComputerName != nil {
+				cached, ok := localCache.Load(*vm.OsProfile.ComputerName)
```

Comment:
what's the value of `vm.OsProfile.ComputerName`? is it equal to k8s nodeName?

Reply:
Yes, vm.OsProfile.ComputerName always equal to K8s nodeName. I use vm.Name previously, however, I discover vmName is not always equal to nodeName. More specifically:
vmName == nodeName applies in the below three cases:
- case 1: using standalone VM
- case 2：using availabilitySet VM
- case 3: using vmss flex and VM is individually created and attached to existing vmss flex

In the below cases, vmName != nodeName
- case 4: using vmss uniform
- case 5: using vmss flex and VM is created by using vmss default VM profile and inscreasing vmss capacity

To satisfy case 3 and case 5 at the same time, we need to use vm.OsProfile.ComputerName

Reply:
in such cases(4,5), k8s only knows `nodeName` in disk attach/detach, how can we hit  `nodeName` in cache? original way is always using nodeName in cache, that could make sure the vmss cache is always working.

Reply:
nodeName == vm.OsProfile.ComputerName applies to case 1 - 5; while nodeName == VMName only applies to case 1 -3. That's why I change to use vm.OsProfile.ComputerName